### ISSUE TITLE
unused/deprecated/unreachable symbols are still reported and greyed out even when the error is suppressed or `typeCheckingMode` is "off"

### ISSUE THREAD

**Issue by DetachHead** [DetachHead/basedpyright#149]
![image](https://github.com/DetachHead/basedpyright/assets/57028336/bc9c1cad-5c36-495f-95f0-4f2cd3293d07)

```toml
[tool.basedpyright]
typeCheckingMode = 'off'
```
```py
def _f(): ...  # message is reported
```

**Comment by DetachHead** [comment]
same with deprecated & unreachable diagnostics

i think the solution is to add those categories to the diagnostics rule options where they're applicable, so the types for them would look something like this:

```ts
interface DiagnosticRules {
    reportAny: DiagnosticCategory // 'error' | 'warning' | 'none'
    reportUnreachable: DiagnosticCategory | 'unreachable'
    reportUnusedVariable: DiagnosticCategory: DiagnosticCategory | 'unused'
}
```

**Comment by DetachHead** [comment]
related: https://github.com/microsoft/pylance-release/discussions/2708
